experimental/SetPartitions/src/LinearSetPartition.jl

```diff
@@ -0,0 +1,241 @@
+"""
+    LinearSetPartition
+
+LinearSetPartition represents a linear combination of set-partitions. See Chapter 5 in [Gro20](@cite).
+"""
+struct LinearSetPartition{S <: AbstractPartition, T <: RingElement}
+    coefficients :: Dict{S, T}
+
+    function LinearSetPartition{S, T}(coeffs::Dict{S, T}) where {S <: AbstractPartition, T <: RingElement}
+        return new(simplify_operation_zero(coeffs))
+    end
+end 
+
+"""
+    linear_partition(coeffs::Dict{S, T}) where {S <: AbstractPartition, T <: RingElement}
+
+Initialize LinearSetPartition object, while simplifying the term.
+
+# Examples
+```jldoctest
+julia> S, d = polynomial_ring(QQ, "d") 
+(Univariate polynomial ring in x over QQ, d)
+julia> linear_partition(Dict(set_partition([1, 2], [1, 1]) => S(4), set_partition([1, 1], [1, 1]) => 4*d))
+LinearSetPartition(Dict(set_partition([1, 2], [1, 1]) => S(4), set_partition([1, 1], [1, 1]) => 4*d))
+"""
+function linear_partition(coeffs::Dict{S, T}) where {S <: AbstractPartition, T <: RingElement}
+    return LinearSetPartition{S, T}(coeffs)
+end
+
+"""
+    coefficients(p::LinearSetPartition{S, T}) where { S <: AbstractPartition, T <: RingElement }
+
+Get the constructor field `coefficients`, the partition term, in form of a `Dict` from
+`SetPartition` to coefficient.
+""" 
+function coefficients(p::LinearSetPartition{S, T}) where { S <: AbstractPartition, T <: RingElement }
```

Comment:
```suggestion
function coefficients(p::LinearSetPartition)
```
at places like this, where each type variable (here `S` and `T`) only occur at a single position (i.e. NOT in multiple types in the signature and NOT in the function body) and you don't restrict their type (the restrictions here are identical to the struct definition), you don't need to give the type parameters at all